src/iris/pipelines/iris_pipeline.py

```diff
@@ -84,6 +84,19 @@ def __init__(
         self.nodes = self.instanciate_nodes()
         self.call_trace = self.env.call_trace_initialiser(nodes=self.nodes, pipeline_nodes=self.params.pipeline)
 
+    def update_config(self, config: str) -> None:
+        """Update the pipeline configuration based on the provided base64-encoded string.
+
+        Args:
+            config (str): Base64-encoded string of the new configuration.
+        """
+        decoded_config_str = base64_decode_str(config)
+        self.params = self.__parameters_type__(**self.load_config(decoded_config_str))
```

Comment:
```suggestion
        config_dict = self.load_config(decoded_config_str)
        self.params = self.__parameters_type__(**config_dict)
```
feels like it is easier to read and debug if it is written like this. It will also give you a clearer traceback, if something fails